app/gui/src/dashboard/App.tsx

```diff
@@ -594,5 +592,49 @@ function LocalBackendPathSynchronizer() {
       localBackend.resetRootPath()
     }
   }
+
   return null
 }
+
+/** Subscribe to queries to keep them up to date. */
+function QuerySubscriber() {
+  useRefetchDirectories(BackendType.local)
+  useRefetchDirectories(BackendType.remote)
+
+  return null
+}
+
+/** Periodically refetch directories for the given backend type. */
+function useRefetchDirectories(backendType: BackendType) {
+  const queryClient = reactQuery.useQueryClient()
+  const enableAssetsTableBackgroundRefresh = useFeatureFlag('enableAssetsTableBackgroundRefresh')
+  const assetsTableBackgroundRefreshInterval = useFeatureFlag(
+    'assetsTableBackgroundRefreshInterval',
+  )
+
+  // We use a different query to refetch the directory data in the background.
+  // This reduces the amount of rerenders by batching them together, so they happen less often.
+  reactQuery.useQuery({
+    queryKey: [backendType, 'refetchListDirectory'],
+    queryFn: async () => {
+      const anyMatchingMutation = queryClient
```

Comment:
Can we use `enabled` for that? looks like a better approach

Reply:
Also React query has a hook for that: https://tanstack.com/query/latest/docs/framework/react/reference/useIsMutating